xcode/Subconscious/Shared/Components/AppView.swift

```diff
@@ -391,7 +396,7 @@ struct AppModel: ModelProtocol {
             return AppAddressBookCursor.update(
                 state: state,
                 action: action,
-                environment: environment.addressBook
+                environment: AddressBookEnvironment(environment)
```

Comment:
Constructing AddressBookEnvironment on the fly instead.

The way I've got this set up at the moment is environments hold services. Services are classes (or maybe actors in future), but environments are just structs to hold these services. This makes them cheap to create and throw away, provided the services are just passed down by reference. I've created a convenience initializer for `AddressBookEnvironment` that takes an `AppEnvironment`.